### PR TITLE
[xmlsec] Update, fix usage

### PR DESCRIPTION
**Pull request by dg0yt**
- #### What does your PR fix?
  Update to 1.2.36.
  Makes unofficial CMake usage explicit.

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
  unchanged, no

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
  yes

---

To be considered: Drop the vendored build system and the unofficial CMake config. (Provide pkg-config files on all platforms.)

### CODE REVIEW

File: ports/xmlsec/fix-size-max.patch [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+ 
+ #include <limits.h>
++#include <stdint.h>
+ #include "errors_helpers.h"
+ 
+-/* if it is missing */
+-#ifndef SIZE_MAX
+-#define SIZE_MAX (~(size_t)0)
+-#endif /* SIZE_MAX */
+-
```
Comment (MonicaLiu0311) [inline ports/xmlsec/fix-size-max.patch:16]: `1.2.35` was released 9 days ago, but I found that the upstream released `1.2.36` 5 days ago, the latest version has fixed this problem, see: https://github.com/lsh123/xmlsec/commit/3445fbfe1aceb88a255b90ee1c06df2d05c188b0.

Comment (dg0yt) [reply ports/xmlsec/fix-size-max.patch:16]: Updated.

Comment (MonicaLiu0311) [reply ports/xmlsec/fix-size-max.patch:16]: Yes, but you didn't remove `fix-size-max.patch`.

Comment (dg0yt) [reply ports/xmlsec/fix-size-max.patch]: Done.

**Review by MonicaLiu0311 — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thank you!

**Pull request merged**